src/App.jsx

```diff
@@ -14,45 +23,142 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+let previousGoods = [...goodsFromServer];
+
+const getSortedGoods = (goods, sortMethod, isReverse) => {
+  let newGoods = [...goods];
+  const isReverseSwitched = isReverse !== previousReverse;
+
+  if (isReverseSwitched) {
+    previousReverse = isReverse;
+  }
+
+  switch (sortMethod) {
+    case SORT_ALPHABETICALLY:
+      newGoods = newGoods.sort((good1, good2) => (isReverse
+        ? good2.localeCompare(good1)
+        : good1.localeCompare(good2)));
+      break;
+    case SORT_BY_LENGTH:
+      newGoods = isReverse
+        ? newGoods.sort()
+          .sort((good1, good2) => good1.length - good2.length).reverse()
+        : newGoods.sort()
+          .sort((good1, good2) => good1.length - good2.length);
+      break;
+    case RESET:
+      newGoods = [...goodsFromServer];
+      break;
+    default:
+      newGoods = isReverseSwitched ? newGoods.reverse() : newGoods;
+  }
+
+  return newGoods;
+};
+
+export const App = () => {
+  const [selectedSort, setSelectedSort] = useState({
+    sortMethod: '',
+    isReverse: false,
+  });
+  const sortedGoods = getSortedGoods(
+    previousGoods,
+    selectedSort.sortMethod,
+    selectedSort.isReverse,
+  );
+
+  previousGoods = sortedGoods;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleOnClick = (event) => {
+    if (event.target.name === selectedSort.sortMethod) {
+      return;
+    }
+
+    if (event.target.name === REVERSE) {
+      setSelectedSort({
+        sortMethod:
+          selectedSort.sortMethod === RESET ? '' : selectedSort.sortMethod,
+        isReverse: !selectedSort.isReverse,
+      });
+    } else {
+      setSelectedSort({
+        sortMethod: event.target.name,
+        isReverse: selectedSort.isReverse,
+      });
+    }
+
+    if (event.target.name === RESET) {
+      setSelectedSort({
+        sortMethod: event.target.name,
+        isReverse: false,
+      });
+    }
+  };
+
+  const hideReset = goodsFromServer.join('') === sortedGoods.join('');
```

Comment:
You can just check if the sortMethod is set and/or isReversed is true instead. Also, it should be clear from the name that the value is boolean
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/96409563/4bc0aafb-c77a-473c-907f-548726605fb0)

Reply:
Fixed that